tests/bluetooth/bsim_bt/bsim_test_adv_resume/src/bs_bt_utils.c

```diff
@@ -0,0 +1,179 @@
+
+void test_tick(bs_time_t HW_device_time)
+{
+	bs_trace_debug_time(0, "Simulation ends now.\n");
+	if (bst_result != Passed) {
+		bst_result = Failed;
+		bs_trace_error("Test did not pass before simulation ended.\n");
+	}
+}
+
+void test_init(void)
+{
+	bst_ticker_set_next_tick_absolute(TEST_TIMEOUT_SIMULATED);
+	bst_result = In_progress;
+}
+
+DEFINE_FLAG(flag_is_connected);
+
+void wait_connected(void)
+{
+	UNSET_FLAG(flag_is_connected);
+	WAIT_FOR_FLAG(flag_is_connected);
+	printk("connected\n");
+}
+
+void wait_disconnected(void)
+{
+	SET_FLAG(flag_is_connected);
+	WAIT_FOR_FLAG_UNSET(flag_is_connected);
+	printk("disconnected\n");
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	UNSET_FLAG(flag_is_connected);
+}
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	if (err != 0) {
+		return;
+	}
+
+	SET_FLAG(flag_is_connected);
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = connected,
+	.disconnected = disconnected,
+};
+
+void bs_bt_utils_setup(void)
+{
+	int err;
+
+	err = bt_enable(NULL);
+	ASSERT(!err, "bt_enable failed.\n");
+}
+
+static bt_addr_le_t last_scanned_addr;
+
+static void scan_connect_to_first_result__device_found(const bt_addr_le_t *addr, int8_t rssi,
+						       uint8_t type, struct net_buf_simple *ad)
+{
+	char addr_str[BT_ADDR_LE_STR_LEN];
+	int err;
+
+	/* We're only interested in connectable events */
+	if (type != BT_HCI_ADV_IND && type != BT_HCI_ADV_DIRECT_IND) {
+		FAIL("Unexpected advertisement type.");
+	}
+
+	bt_addr_le_to_str(addr, addr_str, sizeof(addr_str));
+	printk("Got scan result, connecting.. dst %s, RSSI %d\n", addr_str, rssi);
+
+	err = bt_le_scan_stop();
+	ASSERT(!err, "Err bt_le_scan_stop %d", err);
+
+	struct bt_conn *conn;
+
+	err = bt_conn_le_create(addr, BT_CONN_LE_CREATE_CONN, BT_LE_CONN_PARAM_DEFAULT, &conn);
+	ASSERT(!err, "Err bt_conn_le_create %d", err);
+
+	/* Save address for later comparison */
+	memcpy(&last_scanned_addr, addr, sizeof(last_scanned_addr));
+}
+
+void scan_connect_to_first_result(void)
+{
+	int err;
+
+	printk("start scanner\n");
+	err = bt_le_scan_start(BT_LE_SCAN_PASSIVE, scan_connect_to_first_result__device_found);
+	ASSERT(!err, "Err bt_le_scan_start %d", err);
+}
+
+static void scan_expect_same_address__device_found(const bt_addr_le_t *addr, int8_t rssi,
+						   uint8_t type, struct net_buf_simple *ad)
+{
+	char addr_str[BT_ADDR_LE_STR_LEN];
+	char expected_addr_str[BT_ADDR_LE_STR_LEN];
+
+	/* We're only interested in connectable events */
+	if (type != BT_HCI_ADV_IND && type != BT_HCI_ADV_DIRECT_IND) {
+		FAIL("Unexpected advertisement type.");
+	}
+
+	if (!bt_addr_le_eq(&last_scanned_addr, addr)) {
+		bt_addr_le_to_str(&last_scanned_addr, expected_addr_str, sizeof(expected_addr_str));
+		bt_addr_le_to_str(addr, addr_str, sizeof(addr_str));
+		FAIL("Expected advertiser with addr %s, got %s\n", expected_addr_str, addr_str);
+	}
+	PASS("Advertiser used correct address on resume\n");
```

Comment:
Not a huge fan of doing `PASS` in a file called `utils`. IMO the test case implementation should be responsible for passing/failing tests.

Reply:
yeah utils isn't really utils. I took the structure from `bsim_test_bond_overwrite_denied` because I liked that the test sequence itself (in central.c and peripheral.c) was very clear.
utils is the test case implementation really; the `central.c` and `peripheral.c` are just the abstract test sequence.
I can rename it to `common.c` if you prefer.

Reply:
It's more about having test PASS/FAIL criteria outside the test case implementation, rather than the naming. Renaming to `common` isn't going to help that. 

Having PASS/FAIL criteria outside the test case implementation, will make it harder to reuse for other test cases that may want to use the same common functions.

Reply:
as I said, I consider this file to be part of the test case implementation. It's just split off into its own file for clarity's sake.

I know we love huge files in zephyr, so I can definitely move back all the code into the same `central.c` or `peripheral.c` if we prefer that.

The second sentence is a bit disingenuous, since it's litteraly one line to remove. The chance the function will be re-used as-is is pretty slim, usually we need to make a small tweak, so removing _one_ PASS/FAIL line isn't the end of the world IMO.

Reply:
> as I said, I consider this file to be part of the test case implementation. It's just split off into its own file for clarity's sake.

The reason why it is in a common file, is specifically because you use it in with two test cases, right? You have the peripheral and central tests, which both use it (I realize that I have no been explicitly in saying that those are actually two different test implementations, but they should be considered as such with their own `test_id`s, etc.). 

Anyways, having PASS/FAIl scattered around in multiple files will make it much harder to debug in the future IMO.

Reply:
That's a fair point. I added a comment in the DUT to signal that the PASS will come from the connected callback.
Then it's only FAIL that is sprinkled around, I just used it as an analogue to ASSERT, since that's what was being done when I initially copied this utils file. I can definitely change them to assert statements if you prefer.

Reply:
No, it's alright for now I think